synapse/handlers/message.py

```diff
@@ -906,7 +906,18 @@ def persist_and_notify_client_event(
             directory_handler = self.hs.get_handlers().directory_handler
             if room_alias_str and room_alias_str != original_alias:
                 room_alias = RoomAlias.from_string(room_alias_str)
```

Comment:
we've got two copies of these 20 lines. Factor them out to a func?

Reply:
Yeah, these grew a bit and should be factored out!